### PR TITLE
Followship features

### PR DESCRIPTION
**Pull request by sansuaza**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -26,6 +26,10 @@ gem 'simple_form'
 gem 'devise'
 gem 'cancancan'
 
+gem 'bootstrap', '~> 4.5.0'
+gem 'jquery-rails'
+
```
Comment (juanhurtado10) [inline Gemfile:30]: @sansuaza code consistency (extra spaces) is relevant when we talk about code quality.

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -13,3 +13,6 @@
  *= require_tree .
  *= require_self
  */
+
+ @import "bootstrap";
+
```
Comment (juanhurtado10) [inline app/assets/stylesheets/application.scss]: @sansuaza Please check this out!

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
     redirect_to root_path
   end
+
+
+  protected
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:username, :first_name, :last_name])
+  end
+
+  def followhing_user?(user)
```
Comment (juanhurtado10) [inline app/controllers/application_controller.rb]: @sansuaza is a controller a good place for this method? do you need it in multiple controllers and views? maybe a helper would be a better fit? what do you think?

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    if @user && current_user.following_user?(@user)
+      current_user.unfollow(@user)
+      redirect_to "/users/show/#{@user.username}"
+    end
+  end
+
+  private
+
+  def set_user
+    @user= User.find_by( username: params[:username])
```
Comment (juanhurtado10) [inline app/controllers/users_controller.rb]: @sansuaza do not forget it ... "code consistency"

File: app/models/article.rb [hunk 5]
```diff
@@ -10,4 +10,3 @@ class Article < ApplicationRecord
   # Handle of dont permit that the article dont get the same title
   validates :title, uniqueness: {case_insensitive: false}
 end
-
```
Comment (juanhurtado10) [inline app/models/article.rb:13]: @sansuaza please check this out!

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  #helper_method :current_user_owner?
+
+  def show
+    @user
+  end
+
+  def follow
+    if @user && !current_user.following_user?(@user)
+      current_user.follow(@user)
+      redirect_to "/users/show/#{@user.username}"
```
Comment (juanhurtado10) [inline app/controllers/users_controller.rb]: @sansuaza absolute routes need to be changed by route helpers methods!

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (sansuaza) [reply app/controllers/application_controller.rb]: Yes, definitely thats a good practice, inclusive i can reuse it in the`can_comment?`  method  ubicated in the application helper

File: app/controllers/users_controller.rb (hunk 4 above)
Comment (sansuaza) [reply app/controllers/users_controller.rb]: solved!

File: app/models/article.rb (hunk 5 above)
Comment (sansuaza) [reply app/models/article.rb:13]: solved!

File: app/controllers/users_controller.rb (hunk 6 above)
Comment (sansuaza) [reply app/controllers/users_controller.rb]: solved!

File: Gemfile (hunk 1 above)
Comment (sansuaza) [reply Gemfile:30]: solved!

**Review by juanhurtado10 — APPROVED** [review]

**Pull request merged**